### PR TITLE
backport: trivial 2024 02 22

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
batch of trivial backports; small batch as a lot of them ended up causing test failures

## How Has This Been Tested?
Building locally; tests not ran

## Breaking Changes
Didn't see any yet; didn't review much yet

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: test/functional/p2p_compactblocks.py [hunk 1]
```diff
@@ -561,6 +561,15 @@ def test_getblocktxn_handler(self, test_node):
             assert_equal(test_node.last_message["block"].block.sha256, int(block_hash, 16))
             assert "blocktxn" not in test_node.last_message
 
+        # Request with out-of-bounds tx index results in disconnect
+        bad_peer = self.nodes[0].add_p2p_connection(TestP2PConn())
+        block_hash = node.getblockhash(chain_height)
+        block = from_hex(CBlock(), node.getblock(block_hash, False))
```
Comment (UdjinM6) [inline test/functional/p2p_compactblocks.py:567]: 25370: this should fix linter and tests
```suggestion
        bad_peer = self.nodes[0].add_p2p_connection(TestP2PConn(cmpct_version=1))
        block_hash = node.getblockhash(chain_height)
        block = FromHex(CBlock(), node.getblock(block_hash, False))
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/init.cpp [hunk 2]
```diff
@@ -1370,7 +1370,7 @@ bool AppInitParameterInteraction(const ArgsManager& args)
 
     peer_connect_timeout = args.GetArg("-peertimeout", DEFAULT_PEER_CONNECT_TIMEOUT);
     if (peer_connect_timeout <= 0) {
-        return InitError(Untranslated("peertimeout cannot be configured with a negative value."));
+        return InitError(Untranslated("peertimeout must be a positive integer."));
```
Comment (knst) [inline src/init.cpp:1373]: bitcoin-25506:

consider including 25511 which is tests for it

**Review by knst — APPROVED** [review]
utACK

**Pull request merged**